core/shared/src/main/scala/fs2/internal/Scope.scala

```diff
@@ -478,6 +474,8 @@ private[fs2] final class Scope[F[_]] private (
 
 private[fs2] object Scope {
 
+  private[Scope] val closedScopeException = new RuntimeException("Scope closed at time of lease")
```

Comment:
Is this safe to reuse? Seems like each time it's thrown it will get a new stack trace which then could cause frames to be retained.

Reply:
I am not sure about that. Will keep inline, for good measure.

Reply:
One issue is that `fillInStackTrace` will get called each time this is thrown. But since there was a single instance stored on the object, the last recorded stack trace hangs around forever. Hence if using singleton exceptions, `fillInStackTrace` should be overridden to be a no-op (e.g. by mixing in `NoStackTrace`).